test/e2e/framework/nodes.go

```diff
@@ -50,9 +50,7 @@ func FindGatewayNodes(cluster ClusterIndex) []v1.Node {
 func FindNonGatewayNodes(cluster ClusterIndex) []v1.Node {
 	nodes, err := KubeClients[cluster].CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{
 		LabelSelector: labels.NewSelector().Add(
-			// Ignore the control plane node labeled as master as it doesn't allow scheduling of pods
-			NewRequirement("node-role.kubernetes.io/master", selection.DoesNotExist, []string{}),
-			NewRequirement(GatewayLabel, selection.NotEquals, []string{"true"})).String(),
+			NewRequirement(GatewayLabel, selection.DoesNotExist, []string{})).String(),
```

Comment:
I think we should keep the GW label check as is (as long as it works). The GW criteria is the label set to “true” so conversely the label missing or set to “false” is more complete/correct.

Reply:
Done.
I have verified the fix on "kind" environment with k8s version - 1.23 and 1.25.
Also, I have verified it on ocp based environment.

Reply:
We want just `NotEquals` and not also `DoesNotExist`.